### ISSUE TITLE
Document that embedded Tomcat must be at least 10.1.25

### ISSUE THREAD

**Issue by g00glen00b** [spring-projects/spring-boot#42849]
The [System Requirements](https://docs.spring.io/spring-boot/system-requirements.html) documentation specifies which versions are supported. However, not all patch versions of Tomcat v10.1.x are supported. For example, since #41093, only Tomcat v10.1.25 or later are supported. Running a Spring Boot v3.3.4 application on Tomcat v10.1.24 or lower will result in an error since `AbstractProtocol.setMaxQueueSize(int)` does not exist.

Should the documentation make it more clear that certain patch versions won't work? For example, is it useful to update it to:

> | Name | Servlet Version |
> | ------ | --------------- |
> | Tomcat 10.1 (10.1.25 or later) | 6.0 |
> | Jetty 12.0 | 6.0 |
> | Undertow 2.3 | 6.0 |

Or would this cause too much of a maintenance burden?

**Comment by wilkinsona** [comment]
Out of interest, what's your use case for downgrading Tomcat?

**Comment by g00glen00b** [comment]
I don't have a use case myself. I was answering a question on Stack Overflow of someone who is trying to use an embedded Tomcat in development, but is deploying on an external Tomcat as a WAR file on other environments. That external Tomcat happens to be v10.1.6 and they wanted to stay as close as possible so were overriding the `tomcat.version` property to the same version of their external Tomcat. They also referred to the System requirements documentation, arguing that it should be possible as that page implies that any Tomcat v10.1.x version should be supported. 

For context, here is the Q&A: https://stackoverflow.com/q/79116989

**Comment by wilkinsona** [comment]
Thanks for the background and for helping the community on Stack Overflow.

IMO, they're reading a little too much into that table. In the section on [customizing dependency versions](https://docs.spring.io/spring-boot/how-to/build.html#howto.build.customize-dependency-versions), we warn that "each Spring Boot release is designed and tested against this specific set of third-party dependencies. Overriding versions may cause compatibility issues". That's really what happening here.

That said, usually, we do try to be more tolerant of older versions of Tomcat. Unfortunately, that wasn't possible in this case as falling back to the old way of doing this would reintroduce #40945 and #40957.

In short, I think it's worth noting that we require at least 10.1.25.
